src/parser/expression.js

```diff
@@ -827,13 +827,11 @@ pp.parseObjPropValue = function (prop, startPos, startLoc, isGenerator, isAsync,
 
   if (!prop.computed && prop.key.type === "Identifier") {
     if (isPattern) {
-      let illegalBinding = this.isKeyword(prop.key.name);
-      if (!illegalBinding && this.state.strict) {
-        illegalBinding = reservedWords.strictBind(prop.key.name) || reservedWords.strict(prop.key.name);
-      }
-      if (illegalBinding) {
-        this.raise(prop.key.start, "Binding " + prop.key.name);
+      if (this.isKeyword(prop.key.name)) {
```

Comment:
I guess the case slips through the tokenizer check for `isKeyword`? Leaving this because there's one extra failing test when it's removed.